openlibrary/macros/LoanStatus.html

```diff
@@ -52,52 +57,58 @@
-          <input type="submit" class="cta-btn cta-btn--unavailable" id="waitlist_ebook" value="Join Waitlist"/>
-        </form>
-
-$elif page.get('ocaid') and not page.is_access_restricted() and editions_page:
-    $ viewbook = "//%s/stream/XXX?ref=ol" % bookreader_host()
-    <a href="$viewbook.replace('XXX', page.ocaid)" title="Use BookReader to read online" class="cta-btn cta-btn--available">Read</a>
-
-$elif (not page.get('ocaid') or page.is_access_restricted()) and editions_page:
-    <form>
-      <input type="submit" class="cta-btn" disabled value="No ebook available">
+  $ wlsize = availability.get('num_waitlist', 0)
+  $if waiting_loan:
+    <p class="waitinglist-message">
+      $ spot = ctx.user.get_waiting_loan_for(page)['position']
+      $if spot == 1:
+          You are <strong>next</strong> on the waiting list
+      $else:
+          You are <strong>#$(spot)</strong> of $(wlsize) on the waiting list.
+    </p>
+    <form method="POST" action="$borrow_link" class="leave-waitlist waitinglist-form">
+      <input type="hidden" name="action" value="leave-waitinglist"/>
+      <input type="submit" class="cta-btn" id="unwaitlist_ebook" value="Leave waiting list"/>
+    </form>
+  $else:
+    <p class="waitinglist-message">
+      $("Readers waiting for this title: %s" % wlsize if wlsize else "You'll be next in line.")
+    </p>
+    <form method="POST" action="$borrow_link" class="join-waitlist waitinglist-form">
+      <input type="hidden" name="action" value="join-waitinglist"/>
+      <input type="submit" class="cta-btn cta-btn--unavailable" id="waitlist_ebook" value="Join Waitlist"/>
     </form>
 
+$elif (not ocaid or page.is_access_restricted()) and editions_page:
+  <form>
+    <input type="submit" class="cta-btn" disabled value="No ebook available">
+  </form>
 
-$if editions_page and page.get('ocaid'):
-  $:macros.daisy(page, block_name=block_name)
+$if ocaid and page.is_access_restricted() and availability_status.startswith('borrow') and editions_page:
+  <p>
+    <a class="cta-btn cta-btn--preview" href="#bookPreview" id="bookPreviewButton">Preview</a>
+  </p>
+  <div class="hidden">
+    <div class="floater" id="bookPreview">
+      <div class="book-preview">
+        <div class="floaterHead">
+          <h2>$_('Preview Book')</h2>
+          <a class="floaterShut" data-key="book_preview" data-ol-link-track="book_preview"
+             href="javascript:;" onclick="\$.fn.colorbox.close();">&times;<span class="shift">$_("Close")</span></a>
+        </div>
+        <div id="bookPreviewIframe" class="lazyIframe">
+          <iframe width="100%" height="515"
+                  data-src="https://archive.org/stream/$ocaid?wrapper=false">
+          </iframe>
+        </div>
+        <p class="learn-more">
+          <a href="https://archive.org/details/$ocaid" data-key="book_preview_learn_more"
+             data-ol-link-track="book_preview_learn_more">See more</a>
```

Comment:
I think these should have a hierarchy.
```suggestion
             data-ol-link-track="BookPreview|ArchiveLink">See more</a>
```

Reply:
punting for now, will not fix